hyperspy/drawing/utils.py

```diff
@@ -1869,80 +1874,59 @@ def plot_roi_map(signal, rois=1):
     region at each point in the scan. Therefore, regions of the
     scan where a particular spot is intense will appear bright.
     """
+    if signal._plot is None or not signal._plot.is_active:
+        signal.plot()
+
     sig_dims = len(signal.axes_manager.signal_axes)
     nav_dims = len(signal.axes_manager.navigation_axes)
 
-    if signal.axes_manager.signal_dimension == 0:
-        raise ValueError("The signal must have signal dimension > 0.")
+    if sig_dims not in [1, 2]:
+        raise ValueError("The signal must have signal dimension of 1 or 2.")
 
-    if sig_dims not in [1, 2] or nav_dims not in [1, 2]:
-        warnings.warn(
-            (
-                "This function is only tested for signals with 1 or 2 "
-                "signal and navigation dimensions, not"
-                f" {sig_dims} signal and {nav_dims} navigation."
-            )
-        )
+    if nav_dims == 0:
+        raise ValueError("Navigation dimension must larger than 0.")
 
     if isinstance(rois, int):
         if sig_dims == 1:
             rois = _create_span_roi_group(signal.axes_manager.signal_axes[0], rois)
         elif sig_dims == 2:
             rois = _create_rect_roi_group(*signal.axes_manager.signal_axes, rois)
-        else:
-            raise ValueError(
-                (
-                    "Can only generate default ROIs for signals "
-                    f"with 1 or 2 signal dimensions, not {sig_dims}"
-                    ", try providing an explicit `rois` argument"
-                )
-            )
 
     if len(rois) > 3:
         raise ValueError("Maximum number of spans is 3")
 
     colors = ["red", "green", "blue"]
-    roi_signals = []
     roi_sums = []
-
-    all_sum = signal.nansum()
-    all_sum.plot()
+    axes = [axis.index_in_axes_manager for axis in signal.axes_manager.signal_axes]
 
     for i, roi in enumerate(rois):
         color = colors[i]
 
         # add it to the sum over all positions
-        roi.add_widget(all_sum, color=colors[i])
+        roi.add_widget(signal, axes=axes, color=colors[i])
 
         # create a signal that is the spectral slice of sig
-        roi_signal = hs.interactive(
-            roi,
-            signal=signal,
-            event=roi.events.changed,
-            axes=signal.axes_manager.signal_axes,
-        )
-        roi_signals.append(roi_signal)
-
-        # add up the roi sliced signals to one point per nav position
-        roi_sum = roi_signal.nansum(roi_signal.axes_manager.signal_axes)
-
+        roi_sum = roi(signal).nansum(axes)
         # transpose shape to swap these points per nav into signal points
+        # cap to 2, since hyperspy can't signal dimension higher than 2
+        # take the two first navigation dimension by default
```

Comment:
```suggestion
        # Transpose shape to swap these points per nav into signal points.
        # Cap to 2, since hyperspy can't handle signal dimension higher than 2.
        # Take the two first navigation dimensions by default.
```
```suggestion
        # transpose shape to swap these points per nav into signal points
        # cap to 2, since hyperspy can't signal dimension higher than 2
        # take the two first navigation dimension by default
```